helix-view/src/document.rs

```diff
@@ -939,6 +945,11 @@ impl Document {
         config_loader
             .language_config_for_file_name(self.path.as_ref()?)
             .or_else(|| config_loader.language_config_for_shebang(self.text()))
+            .or_else(|| {
+                self.raw_path
+                    .as_ref()
+                    .and_then(|p| config_loader.language_config_for_file_name(p))
+            })
```

Comment:
If the canonicalized path and shebang don't recognize any language, we can try to guess with the initial path.